test/unit/test_mail.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import tempfile
+from unittest import mock
+from pathlib import Path
+from autosubmit.notifications.mail_notifier import MailNotifier
+from autosubmitconfigparser.config.basicconfig import BasicConfig
+from log.log import Log
+
+@pytest.fixture
+def mock_basic_config():
+    mock_config = mock.Mock()
+    mock_config.MAIL_FROM = "test@example.com"
+    mock_config.SMTP_SERVER = "smtp.example.com"
+    return mock_config
+
+@pytest.fixture
+def mail_notifier(mock_basic_config):
+    return MailNotifier(mock_basic_config)
+
+def test_send_mail(mail_notifier):
+    with mock.patch('smtplib.SMTP') as mock_smtp:
+        mock_server = mock.Mock()
+        mock_smtp.return_value = mock_server
+        mock_server.sendmail = mock.Mock()
+        
+        mail_notifier._send_mail('sender@example.com', 'recipient@example.com', mock.Mock())
+
+        mock_server.sendmail.assert_called_once_with('sender@example.com', 'recipient@example.com', mock.ANY)
+
+def test_attach_files(mail_notifier):
```

Comment:
The functions in this file need to follow pep8 style, especially if it's new code :pray: 

I opened your code in PyCharm, and it immediately displays the number of warnings near the top,

![image](https://github.com/user-attachments/assets/43a666ef-b3ad-462e-9bc6-a401140e3e2d)

Followed by marks on the right sidebar,

![image](https://github.com/user-attachments/assets/ce093e23-0e23-499d-823c-2f3939f72f68)

I will comment the other ones that it found (it normally matches with ruff/pep8/pylint/etc).